server/xcode/xcode_darwin.go

```diff
@@ -134,8 +135,21 @@ func (x *xcodeLocator) locate() error {
 			}
 		}
 	}
-	x.versions = versionMap
-	return nil
+	return versionMap
+}
+
+func xcodePlistForPath(path string) *xcodePlist {
```

Comment:
Can we have xcodePlistForPath return a (*xcodePlist, error), and then check (and potentially log) the error when calling xcodePlistForPath?

This way it's clear when the returned *xcodePlist is valid (no need to check nil), and you can just have a single log.Warningf()